modules/apps/commerce/commerce-product-asset-categories-web/src/main/resources/META-INF/resources/display_layout/edit_category_display_page.jsp

```diff
@@ -150,4 +146,111 @@ if (cpDisplayLayout != null) {
 
 		displayPageItemRemove.classList.add('hide');
 	});
+</aui:script>
+
+<%
+String selectedCategories = StringUtil.merge(assetCategoryIds, StringPool.COMMA);
+%>
+
+<aui:script require="frontend-js-web/liferay/ItemSelectorDialog.es as ItemSelectorDialog">
+	var categoriesContainer = document.querySelector(
+		'#<portlet:namespace />categoriesContainer'
+	);
+
+	function createLabel(id, title) {
+		var labelContainer = document.createElement('span');
+		labelContainer.classList.add('label');
+		labelContainer.classList.add('label-dismissible');
+		labelContainer.classList.add('label-secondary');
+		labelContainer.setAttribute('id', id);
+
+		var linkContainer = document.createElement('span');
+		linkContainer.classList.add('label-item');
+		linkContainer.classList.add('label-item-expand');
+
+		var link = document.createElement('a');
+		link.setAttribute('href', '#' + id);
+		link.innerText = title;
+		linkContainer.appendChild(link);
+
+		var buttonContainer = document.createElement('span');
+		buttonContainer.classList.add('label-item');
+		buttonContainer.classList.add('label-item-after');
+
+		var button = document.createElement('button');
+		button.classList.add('close');
+		button.setAttribute('aria-label', 'Close');
+		button.setAttribute('type', 'button');
+		button.addEventListener('click', handleCloseButtonClick);
+
+		var svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+		svg.classList.add('lexicon-icon');
+		svg.classList.add('lexicon-icon-times');
+		svg.setAttribute('focusable', false);
+		svg.setAttribute('role', 'presentation');
+
+		var use = document.createElementNS('http://www.w3.org/2000/svg', 'use');
+		use.setAttribute(
+			'href',
+			Liferay.ThemeDisplay.getPathThemeImages() + '/clay/icons.svg#times'
+		);
+
+		svg.appendChild(use);
+		button.appendChild(svg);
+		buttonContainer.appendChild(button);
+
+		labelContainer.appendChild(linkContainer);
+		labelContainer.appendChild(buttonContainer);
+
+		return labelContainer;
```

Comment:
@ivicac @julien 

Hello guys, as far as I understand the `add` of the itemSelectorDialog will need to create the element and append it to the `categoriesContainer`, but I'm a bit doubtful about this massive chunk of code. Wouldn'it be better, or do you think it's feasiable to move this into a `JSPF` with the actual HTML structure, and do here something like

```
function createLabel(id, title) {
     var labelContainer = document.createElement('span');
     var labelHTMLContent = '<%@ include file="/itemTemplate.jspf" %>';

     labelContainer.innerHTML = labelHTMLContent;

     var link = labelContainer.querySelector('a.some-link-class');

     link.setAttribute('href', '#' + id);
     link.innerText = title;

     return labelContainer;
}
```

This is an idea, depending on the time you have to edit this.

Reply:
hey @gianmarcobrunialti to be honest I don't know if it's better nor if it's possible (seems possible though) I just wanted to get something working.

On an unrelated note; 40 lines of code doesn't seem that "massive" to me :) and I think that in the future you'll probably want this to be a react component, instead of using AUI

Reply:
Hey @julien! Sorry if my "massive" didn't sound too specific, but it was more related to what the actual function is doing: well, for every addition JS will have create multiple elements with multiple method calls at runtime and then inject them in the DOM. For sure this is also what a JS library like React is doing, but there are other optimizations that are happening there. Considering the template is always the same, I thought it would simply be better to just have it pre-parsed once in a string and injected in the DOM one single time per `createLabel` function call.

But I do agree that if it's just for the sole purpose of having something viable which will then be deprecated, then this shall do as well I guess. :)

Reply:
No worries - I understood what you mean, what I didn't realize is that the template will only have to be parsed once, :+1: thanks for pointing that out